control-base/src/main/java/fi/nls/oskari/control/data/CoordinateTransformation.java

```diff
@@ -0,0 +1,108 @@
+package fi.nls.oskari.control.data;
+
+import fi.nls.oskari.annotation.OskariActionRoute;
+import fi.nls.oskari.control.ActionException;
+import fi.nls.oskari.control.ActionHandler;
+import fi.nls.oskari.control.ActionParameters;
+import fi.nls.oskari.control.ActionParamsException;
+import fi.nls.oskari.domain.geo.Point;
+import fi.nls.oskari.log.LogFactory;
+import fi.nls.oskari.log.Logger;
+import fi.nls.oskari.map.geometry.DefaultPointTransformer;
+import fi.nls.oskari.map.geometry.PointTransformer;
+import fi.nls.oskari.util.IOHelper;
+import fi.nls.oskari.util.JSONHelper;
+import fi.nls.oskari.util.PropertyUtil;
+import fi.nls.oskari.util.ResponseHelper;
+import org.json.JSONArray;
+import org.json.JSONException;
+import org.json.JSONObject;
+
+import java.net.HttpURLConnection;
+import java.util.StringTokenizer;
+
+import static fi.nls.oskari.control.ActionConstants.*;
+
+/**
+ * Transforms coordinates from projection to another. Transformation class can be configured
+ * with property 'projection.library.class' (defaults to fi.nls.oskari.map.geometry.ProjectionHelper).
+ *
+ * Takes 'lan', 'lot', 'srs' and 'targetSRS' parameters and returns a JSONObject with transformed result:
+ * {
+ *     lan: 123,
+ *     lot : 456,
+ *     srs : "EPSG:789"
+ * }
+ */
+@OskariActionRoute("CoordinateTransformation")
+public class CoordinateTransformation extends ActionHandler {
+
+    private static final Logger LOG = LogFactory.getLogger(CoordinateTransformation.class);
+    private static final String URL = "http://193.166.24.38/coordtrans/CoordTrans";
+
+    static final String SOURCE_CRS = "sourceCrs";
+    static final String HEIGHT_CRS = "heightCrs";
+    static final String TARGET_CRS = "targetCrs";
+    static final String COORDINATES = "coords";
+    static final String HEIGHT = "height";
+
+    private PointTransformer service = null;
+
+    @Override
+    public void init() {
+        super.init();
```

Comment:
Overriding the init() function with super.init() doesn't change anything, remove this